sklearn/utils/__init__.py

```diff
@@ -89,6 +90,9 @@
 
 def _array_indexing(array, key, key_dtype, axis):
     """Index an array or scipy.sparse consistently across NumPy version."""
+    xp, is_array_api = get_namespace(array)
+    if is_array_api and not _is_numpy_namespace(xp):
```

Comment:
Indeed, if there is no performance reason (e.g. memory usage with `copy=False` or similar), we should just do:

```suggestion
    if is_array_api:
```

Reply:
Done in 09af4e3e33e3da8504e566edfbb1ce346a4cd410.